api/v2/views/base.py

```diff
@@ -0,0 +1,153 @@
+        status, _ = StatusType.objects.get_or_create(name="pending")
+        try:
+            membership = IdentityMembership.objects.get(identity=identity_id)
+            instance  = serializer.save(
+                membership=membership,
+                status=status,
+                created_by=self.request.user
+            )
+            self.submit_action(instance)
+        except core_exceptions.ProviderLimitExceeded:
+            message = "Only one active request is allowed per provider."
+            raise exceptions.MethodNotAllowed('create', detail=message)
+        except core_exceptions.InvalidMembership:
+            message = (
+                "The user '%s' is not a valid member."
+                % self.request.user.username
+            )
+            raise exceptions.ParseError(detail=message)
+        except IdentityMembership.DoesNotExist:
+            message = (
+                "The identity '%s' does not have a membership"
+                % identity_id
+            )
+            raise exceptions.ParseError(detail=message)
+
+    @unresolved_requests_only
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        self.perform_destroy(instance)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def perform_update(self, serializer):
+        """
+        Updates the request and performs any update actions.
+        """
+        # NOTE: An identity could possible have multiple memberships
+        # It may be better to directly take membership rather than an identity
+        identity_id = serializer.initial_data.get('identity').get("id", None)
+        try:
+            membership = IdentityMembership.objects.get(identity=identity_id)
+            instance = serializer.save(end_date=timezone.now(),
+                                       membership=membership)
+        except core_exceptions.ProviderLimitExceeded:
+            message = "Only one active request is allowed per provider."
+            raise exceptions.MethodNotAllowed('create', detail=message)
+        except core_exceptions.InvalidMembership:
+            message = (
+                "The user '%s' is not a valid member."
+                % self.request.user.username
+            )
+            raise exceptions.ParseError(detail=message)
+        except IdentityMembership.DoesNotExist:
+            message = (
+                "The identity '%s' does not have a membership"
+                % identity_id
+            )
+            raise exceptions.ParseError(detail=message)
+
+        if instance.is_approved():
+            self.approve_action(instance)
+
+        if instance.is_denied():
+            self.deny_action(instance)
+
+    @unresolved_requests_only
+    def update(self, request, *args, **kwargs):
+        """
+        Update the request for the specific identifier
+        """
+        return super(BaseRequestViewSet, self).update(request, *args, **kwargs)
```

Comment:
Is this function calling itself?  It looks like it's defined in `BaseRequestViewSet` and then calls itself, and then calls itself again...?